src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,79 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { addComment } from '../api/posts';
+
+interface Props {
+  postId: number | undefined,
+  getPostInfo: () => void,
+  setError: React.Dispatch<React.SetStateAction<string>>
+}
+
+export const NewCommentForm: React.FC<Props> = ({
+  postId, getPostInfo, setError,
+}) => {
+  const [body, setBody] = useState('');
+  const [email, setEmail] = useState('');
+  const [name, setName] = useState('');
+  const [showRequiredOn, setShowRequiredOn] = useState('');
+
+  const addCommentRequest = () => {
+    if (postId) {
+      if (body && email && name) {
+        setShowRequiredOn('');
+
+        addComment({
+          postId, name, email, body,
+        })
+          .then(() => {
+            getPostInfo();
+            setBody('');
+            setError('');
+          })
+          .catch(() => {
+            setError(
+              'Cannot add comment. '
+              + 'Check your internet connection and try again.',
+            );
+          });
+
+        return;
+      }
+
+      if (!name) {
+        setShowRequiredOn('name');
+
+        return;
+      }
+
+      if (!email) {
+        setShowRequiredOn('email');
+
+        return;
+      }
+
+      if (!body) {
+        setShowRequiredOn('body');
+      }
+    }
+  };
+
+  const onReset = () => {
+    setBody('');
+    setEmail('');
+    setName('');
+  };
+
+  const onSubmit = (ev: React.FormEvent<HTMLFormElement>) => {
```

Comment:
Consider renaming 'onSubmit' to 'handleSubmit' for better readability and understanding of its purpose.